homeassistant/components/yolink/device.py

```diff
@@ -0,0 +1,175 @@
+"""YoLink Device Instance."""
+
+import logging
+from typing import List
+
+from yolink_client.yolink_device import YoLinkDeviceEntry
+from yolink_client.yolink_model import BRDP, BSDPHelper
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+
+class YoLinkDevice(YoLinkDeviceEntry):
```

Comment:
I am not sure why you have some entity related functions like `device_info` and `should_poll` here, while this class does not inherit from Entity.

Reply:
We have several different sensors, YoLinkDeviceEntity is the basic class of our sensor entity
YoLinkSensorEntity
YoLinkBinarySensorEntity
YoLinkBatteryEntity
YoLinkDoorEntity
those entity inherit from YoLinkDeviceEntity, YoLinkDevice class is used to describe device. and this class maintains a list of entities, device_info and should_poll methods are the same for different entities of a device